### ISSUE TITLE
[Bug & workaround] Installing and using PostgreSQL database logs with Lizmap Web Client

### ISSUE THREAD

**Issue by jonnyforestGIS** [3liz/lizmap-web-client#352]
Hi guys,
in file profiles.ini.php
when i change from spatialite profile to pgsql profile i have this:

``` php
default=jauth
jacl2_profile=jauth

[jdb:jauth]
driver="pgsql"
database="lizmap"
host= "localhost"
user= "postgres"
password= "dummypwd"

[jdb:lizlog]
driver="pgsql"
database="lizmap"
host= "localhost"
user= "postgres"
password= "dummypwd"

```

I create successfully the tables to store the logs in postgresql, but when lizmap try to store the informations the postgresql server status give error in the insert of information because the column log_detail have a function DATETIME('now') from sqlite.

As workaround i create a function datetime in lizmap postgresql database like this:

``` sql
CREATE OR REPLACE FUNCTION datetime(now character varying)
  RETURNS timestamp without time zone AS
$BODY$
SELECT CASE 
        WHEN now like 'now' THEN (SELECT cast(now() as timestamp without time zone))
        ELSE NULL
END;
$BODY$
  LANGUAGE sql IMMUTABLE
  COST 100;
```

With this "nasty" function works well and store the informations of log in postgresql database.

What is the file that make the INSERT action? I think this belongs to Jelix infrastructure right?

P.S. I found some "bugs" in documentation that i wrote and i need to add more extra information i will try to add very soon.

**Comment by mdouchin** [comment]
We should add the ability to use PostgreSQL for logs when installing Lizmap, such as for users and groups database

**Comment by pcav** [comment]
+1

**Comment by jonnyforestGIS** [comment]
+1

**Comment by jonnyforestGIS** [comment]
hi i search more about this issue, i think the problem is in here:

 lizmap-web-client/lizmap/modules/lizmap/daos/logDetail.dao.xml

```xml 
<?xml version="1.0" encoding="UTF-8"?>
<dao xmlns="http://jelix.org/ns/dao/1.0">
    <datasources>
        <primarytable name="log_detail" realname="log_detail" primarykey="id" />
    </datasources>
    <record>
        <property name="id" fieldname="id" datatype="integer" autoincrement="true" default=""/>
        <property name="key" fieldname="log_key" datatype="string" required="true"/>
        <property name="timestamp" fieldname="log_timestamp" datatype="datetime" insertpattern="DATETIME('now')"/>
        <property name="user" fieldname="log_user" datatype="string"/>
        <property name="content" fieldname="log_content" datatype="text"/>
        <property name="repository" fieldname="log_repository" datatype="string"/>
        <property name="project" fieldname="log_project" datatype="string"/>
        <property name="ip" fieldname="log_ip" datatype="string"/>
        <!--<property name="" fieldname="" datatype="string/int/float/date"
        required="yes" maxlength="" minlength="" regexp="" sequence=""
        updatepattern="" insertpattern="" selectpattern=""/>-->
    </record>
    <factory>
        <method name="getDetailRange" type="select">
            <parameter name="offset"/>
            <parameter name="limit"/>
            <order>
                <orderitem property="id" way="desc" />
            </order>        
            <limit offset="$offset" count="$limit"/>
        </method>
      
    </factory>
</dao>
```
where is  **insertpattern="DATETIME('now')**
Can you confirm?

**Comment by pieri70** [comment]
Hello
looking into jauth.db I see there is a geobookmark table
as [said here ](https://github.com/3liz/lizmap-web-client/issues/110#issuecomment-277606138) there are some tables that must be created into a given schema.
I was able to create these table by sql statement using set search_path="myschema" at the beginning of my sql

Then I see [here ](https://github.com/3liz/lizmap-web-client/blob/55c0a869a7c8d610d91e1a0faf8fde8d81647488/lizmap/modules/lizmap/install/sql/lizgeobookmark.pgsql.sql) that geobookmark table is created in public schema.
And then it references jlx_user that I imagine should be found in the same schema

> ALTER TABLE public.geobookmark
> ADD CONSTRAINT geobookmark_usr_login_fkey FOREIGN KEY (usr_login)
> REFERENCES jlx_user (usr_login) MATCH SIMPLE
> ON UPDATE CASCADE ON DELETE CASCADE;

Should it be less confusing to create an option in some installation script to point to some given postgresql schema during installation process? This way all the needed tables are stored in this particular schema and not elsewhere around the db?

Thanks

**Comment by rldhont** [comment]
Has been fixed by d893f491aa85f7f05c8a3b664c168ce869d6da40
